From 4d6cfa60f7eff51162525901c17bb6b684f3c853 Mon Sep 17 00:00:00 2001
From: Gaurav Asati <gasati@nvidia.com>
Date: Mon, 15 Feb 2016 13:40:40 +0530
Subject: [PATCH 20/28] Bluetooth: Add Nvidia Custom API.

Add generic Nvidia custom API to send BT specific commands
to stack.

Bug 1726461

Change-Id: I8d8d6433dac825ee9ba2a05a3538174713e390bc
Reviewed-on: http://git-master/r/1011901

Bluetooth: Change NV custom API signature.

Change sendNvCustomCommand signature from byte to String.

Bug 1726461

Change-Id: Iab69cf89cf09b8f5f0ee110db74b62330feffc0c
Reviewed-on: http://git-master/r/1024773

Bluetooth: Change NV custom API signature.

Change sendNvCustomCommand signature from string to byte array

Bug 1726362

Change-Id: I99c54cccc1875da03ea3da8b0b9c1c1a2ee5ae64
Reviewed-on: http://git-master/r/1114016
Reviewed-on: http://git-psac/r/5930
Reviewed-by: Om Prakash Singh <omp@nvidia.com>
Tested-by: Om Prakash Singh <omp@nvidia.com>
Reviewed-by: Eric Chuang <echuang@nvidia.com>
Reviewed-on: http://git-psac/r/7141
Reviewed-by: Roger Ma <roma@nvidia.com>
Tested-by: Roger Ma <roma@nvidia.com>
Reviewed-on: http://git-psac/r/8255
Reviewed-by: David Pearce <dpearce@nvidia.com>
Tested-by: David Pearce <dpearce@nvidia.com>
Reviewed-on: http://git-master/r/1212026
Reviewed-on: http://git-psac/r/9933
Tested-by: Gaurav Asati <gasati@nvidia.com>
Reviewed-by: Gaurav Asati <gasati@nvidia.com>
Reviewed-on: http://git-psac.nvidia.com/r/11676
Reviewed-by: Niranjan Artal <nartal@nvidia.com>
Tested-by: Niranjan Artal <nartal@nvidia.com>
Reviewed-on: http://git-psac.nvidia.com/r/12897
Reviewed-on: https://git-master.nvidia.com/r/1552460
Reviewed-on: https://git-master.nvidia.com/r/1585020
---
 .../java/android/bluetooth/BluetoothAdapter.java | 16 ++++++++++++++++
 core/java/android/bluetooth/IBluetooth.aidl      |  2 ++
 2 files changed, 18 insertions(+)

diff --git a/core/java/android/bluetooth/BluetoothAdapter.java b/core/java/android/bluetooth/BluetoothAdapter.java
index fc3a72482a7..2188c4ea994 100644
--- a/core/java/android/bluetooth/BluetoothAdapter.java
+++ b/core/java/android/bluetooth/BluetoothAdapter.java
@@ -1265,6 +1265,22 @@ public final class BluetoothAdapter {
         return false;
     }
 
+      /**
+      * generic API to be called for sending Nvidia bluetooth spefic commands
+      * it will send cmd with provided data to remote device.
+      * return status of command
+      * @hide
+      */
+     public int sendNvCustomCommand(BluetoothDevice device, int cmd, byte[] data, int bufferSize) {
+         if (getState() != STATE_ON) return -1;
+         try {
+             synchronized(mManagerCallback) {
+                 if (mService != null) return mService.sendNvCustomCommand(device, cmd, data, bufferSize);
+             }
+         } catch (RemoteException e) {Log.e(TAG, "", e);}
+         return -1;
+     }
+
     /**
      * Cancel the current device discovery process.
      * <p>Because discovery is a heavyweight procedure for the Bluetooth
diff --git a/core/java/android/bluetooth/IBluetooth.aidl b/core/java/android/bluetooth/IBluetooth.aidl
index 1d7cfc900e4..ca6dd07e680 100644
--- a/core/java/android/bluetooth/IBluetooth.aidl
+++ b/core/java/android/bluetooth/IBluetooth.aidl
@@ -120,6 +120,8 @@ interface IBluetooth
      */
     oneway void requestActivityInfo(in ResultReceiver result);
 
+    int sendNvCustomCommand(in BluetoothDevice device, int cmd, in byte[] data, int bufferSize);
+
     void onLeServiceUp();
     void onBrEdrDown();
 }
-- 
2.17.1

